Automattic/wp-calypso
# State: Make JPO data layer save more optimistic

We're providing an optimistic UI for the JPO flows, and that is nice. However, our saving in the data layer isn't declared as optimistically as the UI would assume it should be. 

Currently, when saving settings, we'll accept those settings as valid only when the request is successful. This causes inconsistencies, as described by #21868 and particularly the case that @ockham has described. Fixes #21868.

This PR updates our saving data layer to be a little more optimistic, updating the settings immediately. In other words, settings are being updated immediately, and we don't need to wait for the request to complete in order to update the UI.

To test:
* Checkout this branch
* In the 'Site Title' step, make sure you have something saved in the description field.
* Change the description.
* Click the 'Next Step' CTA.
* Click 'Back' quickly.
* Verify the new content is displayed in the description field, and not the previous one.
